sources/packages/web/src/components/aest/students/EditCASSupplierInfoModal.vue

```diff
@@ -0,0 +1,146 @@
+<template>
+  <v-form ref="addCASSupplierForm">
+    <modal-dialog-base
+      title="Update supplier/Site information"
+      :showDialog="showDialog"
+    >
+      <template #content>
+        <error-summary :errors="addCASSupplierForm.errors" />
+        <div class="pb-5">
+          <span class="label-value"
+            >Enter new supplier number and/or site number for this student to
+            maintain consistency with CAS validations.</span
+          >
+        </div>
+        <v-text-field
+          hide-details="auto"
+          label="Supplier number"
+          v-model="formModel.supplierNumber"
+          variant="outlined"
+          :rules="[supplierNumberValidationRule]"
+          @blur="formatSupplierNumber($event)"
+          required
+          class="mb-4"
+        />
+
+        <v-spacer />
+
+        <v-text-field
+          hide-details="auto"
+          label="Site code"
+          v-model="formModel.supplierSiteCode"
+          variant="outlined"
+          :rules="[supplierSiteCodeValidationRule]"
+          @blur="formatSupplierSiteCode($event)"
+          required
+        />
+      </template>
+      <template #footer>
+        <check-permission-role :role="allowedRole">
+          <template #="{ notAllowed }">
+            <footer-buttons
```

Comment:
Please add the same loading strategy used for `src\components\aest\students\modals\TriggerReassessmentModal.vue`.